net/pfSense-pkg-mDNS-Bridge/files/usr/local/www/mdns-bridge.php

```diff
@@ -0,0 +1,517 @@
+<?php
+/*
+ * mdns-bridge.php
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2024 Denny Page
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+require_once("guiconfig.inc");
+require_once("mdns-bridge.inc");
+
+$shortcut_section = 'mdns-bridge';
+
+// Configuration paths
+$package_path = 'installedpackages/mdns-bridge/';
+$path_enable = $package_path . 'enable';
+$path_carp_vhid = $package_path . 'carp_vhid';
+$path_active_interfaces = $package_path . 'active_interfaces';
+$path_global_ip_protocols = $package_path . 'global_ip_protocols';
+$path_global_filter_type = $package_path . 'global_filter_type';
+$path_global_filter_list = $package_path . 'global_filter_list';
+$path_disable_packet_filtering = $package_path . 'disable_packet_filtering';
+$path_interfaces = $package_path . 'interfaces';
+
+// Get the current configuration
+$pconfig['enable'] = config_get_path($path_enable);
+$pconfig['carp_vhid'] = config_get_path($path_carp_vhid);
+$pconfig['active_interfaces'] = explode(',', config_get_path($path_active_interfaces, 'lan'));
+$pconfig['global_ip_protocols'] = config_get_path($path_global_ip_protocols, 'both');
+$pconfig['global_filter_type'] = config_get_path($path_global_filter_type, 'none');
+$pconfig['global_filter_list'] = config_get_path($path_global_filter_list, '');
+$pconfig['disable_packet_filtering'] = config_get_path($path_disable_packet_filtering, false);
+$pconfig['interfaces'] = config_get_path($path_interfaces, []);
+
+// Avahi conflict
+$avahi_enabled = config_get_path('installedpackages/avahi/config/0/enable', false) &&
+		 config_get_path('installedpackages/avahi/config/0/reflection', false);
+
+$available_interfaces = get_configured_interface_with_descr();
+unset($available_interfaces['wan']);
```

Comment:
Don't assume `wan` is any particular kind of interface. If there is something in particular that disqualifies an interface from being available, check for that specifically. See here for an example:
https://github.com/pfsense/FreeBSD-ports/blob/4f80f22a9de3e01f744f34ccf8dd5edfb0790947/net/pfSense-pkg-udpbroadcastrelay/files/usr/local/pkg/udpbroadcastrelay/udpbroadcastrelay.inc#L321

Reply:
The assumption that "wan" is a wan interface is common convention in a lot of the code. It helps users from making mistakes.

Reply:
That assumption comes from the earlier days of the project. It's no longer true - though there are some places where the old code hasn't been fixed yet.

Reply:
I understand that there may be other WAN interfaces, but there isn't a way to detect that. There are a LOT of places that assume 'wan' should be excluded from interface selections. Unless there is a function to call such as is_interface_wan(), I don't see any reasonable way to implement. And since 'wan' will always be a wan interface, I don't see any reason to not exclude it. This is a good practice to help the 80% user from making an egregious mistake.

Reply:
WAN interfaces are simply those that have a gateway directly assigned to them (which is implied for those using DHCP for address assignment). If needed, you may check for that specifically. The interface assigned to 'wan' will not always be a WAN-type interface hence why it's incorrect to assume that in code. And yeah, there are a lot of places that still do this - mostly because they haven't been updated in a decade - which tend to cause bugs and generally break configuration that would otherwise be expected to work.

Reply:
To fix 'wan' across the board, we're going to need a function that performs the check. Having each individual package and internal functions perform that check just isn't going to be practical. I'll make the 'wan' change when it comes around. I've already changed 'lan' because it doesn't do any harm.

Reply:
There are two functions used for this now (that is, to check if it's a "WAN-type" interface). You can do the following:
```
$interface = 'wan';
$interface_is_wan = false;
if (interface_has_gateway($interface) || interface_has_gatewayv6($interface)) {
	$interface_is_wan = true;
}
```

Reply:
Okay, I can work with that.